src/shared/popups/gokb-kbart-import-popup/gokb-kbart-import-popup.vue

```diff
@@ -328,29 +331,51 @@
         }
       },
       doImport () {
-        this.error = undefined
-        this.charsetError = false
+        this.errors = []
         this.importRunning = true
         this.completion = 0
         this.charsetReader = new FileReader()
-        this.charsetReader.onload = this._checkCharset
+        this.charsetReader.onload = this._checkFile
 
         this.charsetReader.readAsBinaryString(this.options.selectedFile)
 
         this.readerForImport = new FileReader()
         this.readerForImport.onload = this._importCompleted
         this.readerForImport.onprogress = this._importProgress
 
-        this.readerForImport.readAsText(this.options.selectedFile)
+        if (this.errors?.length == 0){
+          this.readerForImport.readAsText(this.options.selectedFile)
+        }
+      },
+      _checkFile () {
+        this._checkCharset()
+        this._checkLinesLength()
       },
       _checkCharset () {
-        const csvResult = this.charsetReader.result.split(/\r|\n|\r\n/)
+        const csvResult = this.charsetReader.result.split(/\r\n|\n|\r/)
         const encoding = jschardet.detect(csvResult.toString()).encoding
 
         if (encoding !== 'UTF-8' && encoding !== 'ascii') {
           console.log(jschardet.detectAll(csvResult.toString()))
-          this.charsetError = true
-          this.error = this.$i18n.t('kbart.errors.encoding')
+          this.errors.push(this.$i18n.t('kbart.errors.encoding'))
+        }
+      },
+      _checkLinesLength () {
+        const csvDataRows = this.charsetReader.result.split(/\r?\n/)
+          .filter(row => row.trim())
+        if (csvDataRows.length < 2) {
+          return
+        }
+        const columnsCount = csvDataRows[0].split(/\t/).length
+        const wrongColumnSizes = new Map()
+        csvDataRows.forEach(function (row, i) {
+          var rowLength = row.split(/\t/).length
+          if (rowLength != columnsCount)
+          wrongColumnSizes.set(i, rowLength)
+        });
+        if (wrongColumnSizes.size != 0) {
+          console.log('kbart.errors.containsDifferentLengthedRows :', this.$i18n.t('kbart.errors.containsDifferentLengthedRows'))
```

Comment:
Don't think this log is necessary any more.

Reply:
Removed.